securebanking-openbanking-uk-rs-simulator/src/test/java/com/forgerock/securebanking/openbanking/uk/rs/api/obie/event/v3_1_2/callbackurl/CallbackUrlsApiControllerTest.java

```diff
@@ -0,0 +1,181 @@
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private CallbackUrlsRepository callbackUrlsRepository;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @AfterEach
+    void removeData() {
+        callbackUrlsRepository.deleteAll();
+    }
+
+    @Test
+    public void shouldCreateCallbackUrl() {
+        // Given
+        OBCallbackUrl1 obCallbackUrl1 = aValidOBCallbackUrl1();
+        String tppId = UUID.randomUUID().toString();
+        HttpEntity<OBCallbackUrl1> request = new HttpEntity<>(obCallbackUrl1, requiredEventHttpHeaders(callbacksUrl(), tppId));
+
+        // When
+        ResponseEntity<OBCallbackUrlResponse1> response = restTemplate.postForEntity(callbacksUrl(), request, OBCallbackUrlResponse1.class);
+
+        // Then
+        assertThat(response.getStatusCode()).isEqualTo(CREATED);
+        OBCallbackUrlResponseData1 responseData = response.getBody().getData();
+        assertThat(responseData.getCallbackUrlId()).isNotNull();
+        assertThat(responseData.getUrl()).isEqualTo(obCallbackUrl1.getData().getUrl());
+        assertThat(responseData.getVersion()).isEqualTo(obCallbackUrl1.getData().getVersion());
+        assertThat(response.getBody().getMeta()).isNotNull();
+        // TODO - enable as part of #54
+        //assertThat(response.getBody().getLinks().getFirst().equals(callbacksUrl()));
+    }
+
+    @Test
+    public void shouldReadCallBackUrls() {
+        // Given
+        OBCallbackUrl1 obCallbackUrl1 = aValidOBCallbackUrl1();
+        String callbacksUrl = callbacksUrl();
+        HttpHeaders headers = requiredEventHttpHeaders(callbacksUrl, UUID.randomUUID().toString());
+        HttpEntity<OBCallbackUrl1> request = new HttpEntity<>(obCallbackUrl1, headers);
+        ResponseEntity<OBCallbackUrlResponse1> persistedCallback = restTemplate.postForEntity(callbacksUrl, request, OBCallbackUrlResponse1.class);
+
+        // When
+        ResponseEntity<OBCallbackUrlsResponse1> response = restTemplate.exchange(callbacksUrl, GET, new HttpEntity<>(headers), OBCallbackUrlsResponse1.class);
+
+        // Then
+        assertThat(response.getStatusCode()).isEqualTo(OK);
+        assertThat(response.getBody().getData().getCallbackUrl()).isNotEmpty();
+        OBCallbackUrlResponseData1 callbackUrlResponse = response.getBody().getData().getCallbackUrl().get(0);
+        assertThat(callbackUrlResponse.getCallbackUrlId()).isEqualTo(persistedCallback.getBody().getData().getCallbackUrlId());
+        assertThat(callbackUrlResponse.getUrl()).isEqualTo(obCallbackUrl1.getData().getUrl());
+        assertThat(callbackUrlResponse.getVersion()).isEqualTo(obCallbackUrl1.getData().getVersion());
+        assertThat(response.getBody().getMeta()).isNotNull();
+        // TODO - enable as part of #54
+        //assertThat(response.getBody().getLinks().getFirst().equals(callbacksUrl));
+    }
+
+    @Test
+    public void shouldUpdateCallbackUrl() {
```

Comment:
I struggled to get this to work, hitting a few issues along the way. For reasons I don't fully understand yet, I had to change `isNew()` within the `FRCallbackUrl` Document to check its internal ID, rather than its created date.